prow/cmd/peribolos/main.go

```diff
@@ -409,8 +415,11 @@ func configureTeams(client teamClient, orgName string, orgConfig org.Config) (ma
 	}
 
 	// Delete undeclared teams.
-	// TODO(fejta): consider ensuring we don't delete too many teams at once
-	unused := ints.Difference(used)
+	if reused := unused.Intersection(used); len(reused) > 0 {
+		// This should not never if github is sane.
```

Comment:
Can you provide more detail on the behavior you are trying to catch here? It is not clear to me what case this is handling.

Reply:
> should not never

uh... Clearly a brain-fart in that comment. Let me clean it up.

> more detail

In a serial, single-worker system we can assume that `client.CreateTeam()` will not return any ID X in use by `client.ListTeams()` before we call `client.DeleteTeam(X)`. This is not a safe assumption in a distributed, concurrent system. Therefore we need to ensure that none of the IDs returned by `client.CreateTeam()` are in the `unused` set.

Consider the following (unlikely but logically possible) scenario:
1) Client A calls `client.ListTeam()`, which includes team T
2) Client B calls `client.DeleteTeam(T)`
3) Client A computes `unused := ints.Difference(used)`, which includes includes team T
4) Client A calls `client.CreateTeam()` and github returns team T since this ID is no longer in use
5) Client A calls `client.DeleteTeam(T)` because it is in the unused set

Step 5 relies on the assumption that step 4 will not return a live ID found by step 1, but this is a bad assumption because we are ignorant of whether any step 2s occur between steps 1 and 4.

PS: this is eventually consistent for the scenario where client A and B (both working with the same config) call `client.CreateTeam()` to satisfy team FOO. We will temporarily wind up with IDs X and Y for team FOO but the next time they run they will both agree to delete Y since ID Y > ID X.

PPS: I recognize these are all edge cases but not terribly hard to deal with once we have it in mind.